clang/lib/Sema/SemaType.cpp

```diff
@@ -4953,9 +4953,11 @@ static TypeSourceInfo *GetFullTypeForDeclarator(TypeProcessingState &state,
         // OpenCL v2.0 s6.12.5 - A block cannot be the return value of a
         // function.
         if (T->isBlockPointerType() || T->isImageType() || T->isSamplerT()) {
-          S.Diag(D.getIdentifierLoc(), diag::err_opencl_invalid_return)
-              << T << 1 /*hint off*/;
-          D.setInvalidType(true);
+          if (!T->isSampledImageType()) {
```

Comment:
Very minor style nit: probably this `if` should be folded with previous on similar to `clang/lib/Sema/SemaDecl.cpp`.

Reply:
Yes, made it a single if statement